monitoring/uss_qualifier/scenarios/astm/utm/dss/subscription_validation.py

```diff
@@ -0,0 +1,212 @@
+from datetime import datetime, timedelta
+from typing import Dict, Any, List, Optional
+
+import loguru
+import s2sphere
+from s2sphere import LatLngRect
+from uas_standards.astm.f3548.v21.api import Subscription
+from uas_standards.astm.f3548.v21.constants import Scope
+
+from monitoring.monitorlib.geo import Polygon, Volume3D
+from monitoring.monitorlib.geotemporal import Volume4D
+from monitoring.monitorlib.mutate.scd import MutatedSubscription
+from monitoring.prober.infrastructure import register_resource_type
+from monitoring.uss_qualifier.resources import VerticesResource
+from monitoring.uss_qualifier.resources.astm.f3548.v21.dss import DSSInstanceResource
+from monitoring.uss_qualifier.resources.interuss.id_generator import IDGeneratorResource
+from monitoring.uss_qualifier.resources.uss_area import (
+    USSAreaResource,
+    USSAreaSpecification,
+)
+from monitoring.uss_qualifier.scenarios.astm.utm.dss import test_step_fragments
+from monitoring.uss_qualifier.scenarios.scenario import (
+    TestScenario,
+    PendingCheck,
+)
+from monitoring.uss_qualifier.suites.suite import ExecutionContext
+
+_24H_MIN_TOLERANCE_S = 23 * 3600 + 59 * 60  # 23 hours and 59 minutes
+_24H_MAX_TOLERANCE_S = 24 * 3600 + 1  # 24 hours sharp, plus a second
+
+
+class SubscriptionValidation(TestScenario):
+    """
+    A scenario that verifies that the DSS properly validates and sets the proper limits on subscriptions.
+    """
+
+    SUB_TYPE = register_resource_type(378, "Subscription")
+
+    # Base identifier for the subscriptions that will be created
+    _sub_id: str
+
+    _uss_area: USSAreaSpecification
+
+    _uss_area_volume4d: Volume4D
+
+    # Base parameters used for subscription creation variations
+    _sub_generation_params: Dict[str, Any]
```

Comment:
Since this data structure is essentially untyped (can't tell what members it should have or the types of those members), let's provide that information as documentation:

```suggestion
    # Base parameters used for subscription creation variations
    _sub_generation_params: Dict[str, Any]
    """Parameters used to create subscriptions.
    `subscription_id`: ID of F3548-21 subscription to create/modify.
    `notify_for_op_intents`: True to receive notifications for operational intent activity.
    `notify_for_constraints`: True to receive notifications for constraint activity.
    """
```